gazelle/python/testdata/type_checking_imports/BUILD.in

```diff
@@ -0,0 +1 @@
+# gazelle:python_generation_mode file
```

Comment:
I think we'll need to add test cases for package and project generation mode too :face_with_diagonal_mouth:

Reply:
That was a good call - added it, and it actually uncovered an issue: with multiple files on the same target, if the same import was a type-checking import in one file and a regular import in another file, it could end up in `deps` or in `pyi_deps` depending on the order that the files were processed. This was due to all files being processed in `parser.go` `parse()` function, which parses all files and returns a single treeset. I have fixed the case that affected the test with the new `addModuleToTreeSet` function to force the non-type-checking deps always take precedence over type-checking deps.